From fe2be5a833354c7b280be9c705bef9d11891a422 Mon Sep 17 00:00:00 2001
From: Xiao Chen <xiao@apache.org>
Date: Thu, 2 Feb 2017 10:08:27 -0800
Subject: [PATCH 2198/2848] HDFS-11363. Need more diagnosis info when seeing
 Slow waitForAckedSeqno.

(cherry picked from commit 0914fcca312b5e9d20bcf1b6633bc13c9034ba46)
(cherry picked from commit b0a1703f102d4fd7abc43cce3e5f30a36cda4dcc)

 Conflicts:
	hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DataStreamer.java

Change-Id: Id5820e56b8666626bdf107ec41966b70c137194c
---
 .../org/apache/hadoop/hdfs/DFSOutputStream.java    |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
index 5e38a83..ee69deb 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
@@ -2528,7 +2528,9 @@ private void waitForAckedSeqno(long seqno) throws IOException {
       long duration = Time.monotonicNow() - begin;
       if (duration > dfsclientSlowLogThresholdMs) {
         DFSClient.LOG.warn("Slow waitForAckedSeqno took " + duration
-            + "ms (threshold=" + dfsclientSlowLogThresholdMs + "ms)");
+            + "ms (threshold=" + dfsclientSlowLogThresholdMs
+            + "ms). File being written: " + src + ", block: " + streamer.block
+            + ", Write pipeline datanodes: " + Arrays.toString(streamer.nodes));
       }
     } finally {
       scope.close();
-- 
1.7.9.5

